logfire/_internal/exporters/fallback.py

```diff
@@ -18,9 +19,15 @@ def __init__(
     def export(self, spans: Sequence[ReadableSpan]) -> SpanExportResult:
         try:
             res = self.exporter.export(spans)
-        except Exception:
-            self.fallback.export(spans)
-            raise
+        except Exception as e:
+            if isinstance(e, requests.exceptions.RequestException):
+                # Silence the exception so that OTEL doesn't log a huge traceback.
+                # Rely on OTLPExporterHttpSession to log this kind of error periodically.
+                return SpanExportResult.FAILURE
+            else:
+                # Only write to fallback file if this isn't already being retried by OTLPExporterHttpSession.
+                self.fallback.export(spans)
+                raise
```

Comment:
This code looks good but I don't think it should live in FallbackExporter. It would be better as `SilencingProcessor` or similar.

Reply:
I disagree because we're specifically preventing the exception from resulting in writing to a second fallback file. The extra exporter would be coupled to FallbackExporter and need to be inside.